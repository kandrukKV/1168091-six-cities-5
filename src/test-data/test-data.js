export const noon = () => {};

export const additions = [
  `addition1`,
  `addition2`,
  `addition3`,
  `addition4`,
  `addition5`,
];

export const cities = [
  `Amsterdam`,
  `Paris`,
];

export const mockOffers = [
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: false,
    isPremium: false,
    rating: 1,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.385540000000006,
      longitude: 4.902976,
      zoom: 16
    },
    id: 1
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: true,
    isPremium: true,
    rating: 2,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.388540000000006,
      longitude: 4.899976,
      zoom: 16
    },
    id: 2
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: false,
    isPremium: true,
    rating: 3,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.364540000000005,
      longitude: 4.9019759999999994,
      zoom: 16
    },
    id: 3
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: true,
    isPremium: false,
    rating: 4,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.385540000000006,
      longitude: 4.886976,
      zoom: 16
    },
    id: 4
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: true,
    isPremium: true,
    rating: 5,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.397540000000006,
      longitude: 4.9099759999999995,
      zoom: 16
    },
    id: 5
  }
];

export const mosckReviews = [
  {
    id: 1,
    user: {
      id: 18,
      name: `Sophie`,
      avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`
    },
    rating: 3,
    comment: `We loved it so much, the house, the veiw, the location just great.. Highly reccomend :)`,
    date: `2020-11-02T13:38:44.752Z`
  },
  {
    id: 2,
    user: {
      id: 16,
      name: `Mollie`,
      avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg`
    },
    rating: 2,
    comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`,
    date: `2020-11-02T13:38:44.752Z`
  }
];
