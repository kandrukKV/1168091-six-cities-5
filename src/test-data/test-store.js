import configureStore from "redux-mock-store";
import {mockOffers, mockReviews, cities} from "../test-data/test-data";

const createStore = configureStore();

const mockStore = createStore({
  API_DATA: {
    offers: mockOffers,
    favoriteOffers: mockOffers,
    offerDetails: {
      card: mockOffers[0],
      reviews: mockReviews,
      nearPlaces: mockOffers
    }
  },
  APP_STATE: {
    currentCity: `Amsterdam`,
    cities,
    currentSortType: `Popular`,
    activeCard: null,
    reviewFormState: `EDITING`
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
    authInfo: null
  }
});

export {mockStore};
