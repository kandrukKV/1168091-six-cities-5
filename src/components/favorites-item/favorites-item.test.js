import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import FavoritesItem from "./favorites-item";
import {mockOffers} from "../../test-data/test-data";

const favoritesItem = {
  cityName: `City1`,
  offers: mockOffers
};

it(`FavoritesItem is rendered correctly`, () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(
      <FavoritesItem
        favoritesItem={favoritesItem}
      />
  );
  expect(tree).toMatchSnapshot();
});
