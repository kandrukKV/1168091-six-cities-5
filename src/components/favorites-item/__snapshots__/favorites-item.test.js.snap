// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesItem is rendered correctly 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link"
        href="#"
      >
        <span>
          City1
        </span>
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <PlaceCard
      card={
        Object {
          "additions": Array [
            "Dishwasher",
            "Dishwasher",
            "Dishwasher",
          ],
          "adults": 7,
          "bedrooms": 3,
          "city": Object {
            "location": Object {
              "latitude": 52.37454,
              "longitude": 4.897976,
              "zoom": 13,
            },
            "name": "Amsterdam",
          },
          "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
          "host": Object {
            "avatar": "img/avatar-angelina.jpg",
            "id": 25,
            "name": "Angelina",
          },
          "id": 1,
          "images": Array [
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
          ],
          "isFavorite": false,
          "isPremium": false,
          "location": Object {
            "latitude": 52.385540000000006,
            "longitude": 4.902976,
            "zoom": 16,
          },
          "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
          "price": 440,
          "rating": 1,
          "title": "Perfectly located Castro",
          "type": "house",
        }
      }
      imageHeight={110}
      imageWidth={150}
      itemClassName="favorites__card"
      onMouseOverHandler={[Function]}
      wrapClassName="favorites"
    />
    <PlaceCard
      card={
        Object {
          "additions": Array [
            "Dishwasher",
            "Dishwasher",
            "Dishwasher",
          ],
          "adults": 7,
          "bedrooms": 3,
          "city": Object {
            "location": Object {
              "latitude": 52.37454,
              "longitude": 4.897976,
              "zoom": 13,
            },
            "name": "Amsterdam",
          },
          "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
          "host": Object {
            "avatar": "img/avatar-angelina.jpg",
            "id": 25,
            "name": "Angelina",
          },
          "id": 2,
          "images": Array [
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
          ],
          "isFavorite": true,
          "isPremium": true,
          "location": Object {
            "latitude": 52.388540000000006,
            "longitude": 4.899976,
            "zoom": 16,
          },
          "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
          "price": 440,
          "rating": 2,
          "title": "Perfectly located Castro",
          "type": "house",
        }
      }
      imageHeight={110}
      imageWidth={150}
      itemClassName="favorites__card"
      onMouseOverHandler={[Function]}
      wrapClassName="favorites"
    />
    <PlaceCard
      card={
        Object {
          "additions": Array [
            "Dishwasher",
            "Dishwasher",
            "Dishwasher",
          ],
          "adults": 7,
          "bedrooms": 3,
          "city": Object {
            "location": Object {
              "latitude": 52.37454,
              "longitude": 4.897976,
              "zoom": 13,
            },
            "name": "Amsterdam",
          },
          "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
          "host": Object {
            "avatar": "img/avatar-angelina.jpg",
            "id": 25,
            "name": "Angelina",
          },
          "id": 3,
          "images": Array [
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
          ],
          "isFavorite": false,
          "isPremium": true,
          "location": Object {
            "latitude": 52.364540000000005,
            "longitude": 4.9019759999999994,
            "zoom": 16,
          },
          "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
          "price": 440,
          "rating": 3,
          "title": "Perfectly located Castro",
          "type": "house",
        }
      }
      imageHeight={110}
      imageWidth={150}
      itemClassName="favorites__card"
      onMouseOverHandler={[Function]}
      wrapClassName="favorites"
    />
    <PlaceCard
      card={
        Object {
          "additions": Array [
            "Dishwasher",
            "Dishwasher",
            "Dishwasher",
          ],
          "adults": 7,
          "bedrooms": 3,
          "city": Object {
            "location": Object {
              "latitude": 52.37454,
              "longitude": 4.897976,
              "zoom": 13,
            },
            "name": "Amsterdam",
          },
          "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
          "host": Object {
            "avatar": "img/avatar-angelina.jpg",
            "id": 25,
            "name": "Angelina",
          },
          "id": 4,
          "images": Array [
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
          ],
          "isFavorite": true,
          "isPremium": false,
          "location": Object {
            "latitude": 52.385540000000006,
            "longitude": 4.886976,
            "zoom": 16,
          },
          "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
          "price": 440,
          "rating": 4,
          "title": "Perfectly located Castro",
          "type": "house",
        }
      }
      imageHeight={110}
      imageWidth={150}
      itemClassName="favorites__card"
      onMouseOverHandler={[Function]}
      wrapClassName="favorites"
    />
    <PlaceCard
      card={
        Object {
          "additions": Array [
            "Dishwasher",
            "Dishwasher",
            "Dishwasher",
          ],
          "adults": 7,
          "bedrooms": 3,
          "city": Object {
            "location": Object {
              "latitude": 52.37454,
              "longitude": 4.897976,
              "zoom": 13,
            },
            "name": "Amsterdam",
          },
          "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
          "host": Object {
            "avatar": "img/avatar-angelina.jpg",
            "id": 25,
            "name": "Angelina",
          },
          "id": 5,
          "images": Array [
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
          ],
          "isFavorite": true,
          "isPremium": true,
          "location": Object {
            "latitude": 52.397540000000006,
            "longitude": 4.9099759999999995,
            "zoom": 16,
          },
          "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
          "price": 440,
          "rating": 5,
          "title": "Perfectly located Castro",
          "type": "house",
        }
      }
      imageHeight={110}
      imageWidth={150}
      itemClassName="favorites__card"
      onMouseOverHandler={[Function]}
      wrapClassName="favorites"
    />
  </div>
</li>
`;
